src/main/scala/djinni/YamlGenerator.scala

```diff
@@ -192,46 +193,54 @@ class YamlGenerator(spec: Spec) extends Generator(spec) {
 }
 
 object YamlGenerator {
-  def metaFromYaml(td: ExternTypeDecl) = MExtern(
+  def metaFromYaml(td: ExternTypeDecl, spec: Spec) = MExtern(
     td.ident.name.stripPrefix(td.properties("prefix").toString), // Make sure the generator uses this type with its original name for all intents and purposes
     td.params.size,
     defType(td),
     td.body,
     MExtern.Cpp(
-      nested(td, "cpp")("typename").toString,
-      nested(td, "cpp")("header").toString,
-      nested(td, "cpp")("byValue").asInstanceOf[Boolean]),
+      nested(td, spec.cppOutFolder.isDefined, "cpp", "typename", _.toString),
+      nested(td, spec.cppOutFolder.isDefined, "cpp", "header", _.toString),
+      nested(td, spec.cppOutFolder.isDefined, "cpp", "byValue", _.asInstanceOf[Boolean])),
     MExtern.Objc(
-      nested(td, "objc")("typename").toString,
-      nested(td, "objc")("header").toString,
-      nested(td, "objc")("boxed").toString,
-      nested(td, "objc")("pointer").asInstanceOf[Boolean],
-      nested(td, "objc")("hash").toString),
+      nested(td, spec.objcOutFolder.isDefined, "objc", "typename", _.toString),
+      nested(td, spec.objcOutFolder.isDefined, "objc", "header", _.toString),
+      nested(td, spec.objcOutFolder.isDefined, "objc", "boxed", _.toString),
+      nested(td, spec.objcOutFolder.isDefined, "objc", "pointer", _.asInstanceOf[Boolean]),
+      nested(td, spec.objcOutFolder.isDefined, "objc", "hash", _.toString)),
     MExtern.Objcpp(
-      nested(td, "objcpp")("translator").toString,
-      nested(td, "objcpp")("header").toString),
+      nested(td, spec.objcppOutFolder.isDefined, "objcpp", "translator", _.toString),
+      nested(td, spec.objcppOutFolder.isDefined, "objcpp", "header", _.toString)),
     MExtern.Java(
-      nested(td, "java")("typename").toString,
-      nested(td, "java")("boxed").toString,
-      nested(td, "java")("reference").asInstanceOf[Boolean],
-      nested(td, "java")("generic").asInstanceOf[Boolean],
-      nested(td, "java")("hash").toString,
-      if (nested(td, "java") contains "writeToParcel") nested(td, "java")("writeToParcel").toString else "%s.writeToParcel(out, flags)",
-      if (nested(td, "java") contains "readFromParcel") nested(td, "java")("readFromParcel").toString else "new %s(in)"),
+      nested(td, spec.javaOutFolder.isDefined, "java", "typename", _.toString),
+      nested(td, spec.javaOutFolder.isDefined, "java", "boxed", _.toString),
+      nested(td, spec.javaOutFolder.isDefined, "java", "reference", _.asInstanceOf[Boolean]),
+      nested(td, spec.javaOutFolder.isDefined, "java", "generic", _.asInstanceOf[Boolean]),
+      nested(td, spec.javaOutFolder.isDefined, "java", "hash", _.toString),
+      nested(td, false, "java", "writeToParcel", _.toString) getOrElse "%s.writeToParcel(out, flags)",
+      nested(td, false, "java", "readFromParcel", _.toString) getOrElse "new %s(in)"),
```

Comment:
Fields not required. Here's where we set their default values when not specified.